### PR TITLE
Support custom token

### PR DESCRIPTION
**Pull request by davidkoller318**
Fixed issue https://github.com/polkaswitch/polkaswitch-ui/issues/6

Please review it.
Thank you.

### CODE REVIEW

File: src/client/js/utils/wallet.js [hunk 1]
```diff
@@ -113,6 +109,57 @@ window.WalletJS = {
     return await contract.balanceOf(this.currentAddress());
   },
 
+  getName: function(tokenAddr) {
+    if (this.isConnected() && tokenAddr) {
```
Comment (chillvybz) [inline src/client/js/utils/wallet.js:113]: you don't need to structure this in 2 methods (getName and getERC20Name).
One method is enough.

i.e
```
getDecimals: function(tokenAddr) {
    if (this.isConnected() && tokenAddr) {
      const contract = new Contract(
          tokenAddr,
          window.erc20Abi,
          this.getProvider()
      );
      return contract.decimals();
    } else {
      return Promise.reject();
    }
  },
```

Can you simplify the same with getDecimals and getSymbol

File: src/client/js/utils/wallet.js [hunk 2]
```diff
@@ -49,21 +49,17 @@ window.WalletJS = {
   initListeners: function(provider) {
     provider.on('accountsChanged', function (accounts) {
       // Time to reload your interface with accounts[0]!
-      console.log(accounts);
```
Comment (chillvybz) [inline src/client/js/utils/wallet.js:52]: Keep these logs - just for debugging purposes.

File: src/client/css/custom/swap.scss [hunk 3]
```diff
@@ -449,6 +449,48 @@ $swap-widget-anim-timing: 300ms;
     }
   }
 
+  .warning-view {
```
Comment (chillvybz) [inline src/client/css/custom/swap.scss:452]: rename as `custom-token-warning`. "warning-view" is too generic.

File: src/client/js/components/partials/CustomTokenModal.jsx [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+      fetchingTokenInfo: false,
+    });
+  }
+
+  handleCustomBtn = (e) => {
+    let network = TokenListManager.getCurrentNetworkConfig();
+    const chainId = network.chainId;
+    if (chainId > 0) {
+      const { symbol, name, decimals, customTokenAddr } = this.state
+      const customTokenAddresses = store.get('customTokenAddress') || {};
```
Comment (chillvybz) [inline src/client/js/components/partials/CustomTokenModal.jsx:70]: Can you abstract/move this into `TokenListManager`? So it's all in one place.

So you are calling `TokenListManager.addCustomToken(customToken)`

File: src/client/js/components/partials/CustomTokenModal.jsx [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+        isCustomAddr: true,
+        address: customTokenAddr,
+        chainId: Number(chainId)
+      }
+
+      if (!_.isEmpty(customTokenAddresses) && (!_.isUndefined(customTokenAddresses[chainId]))) {
+        addresses = customTokenAddresses[chainId];
+      }
+      addresses.push(customToken);
+      store.set('customTokenAddress', {[chainId]: addresses});
```
Comment (chillvybz) [inline src/client/js/components/partials/CustomTokenModal.jsx:85]: Can you abstract/move this into `TokenListManager`? So it's all in one place.

So you are calling `TokenListManager.addCustomToken(customToken)`

File: src/client/js/utils/tokenList.js [hunk 6]
```diff
@@ -50,6 +47,7 @@ window.TokenListManager = {
       'fast', 'fastest', 'safeLow'
     ]);
     window.TOKEN_LIST = tokenList;
+    this.updateTokenListwithCustom(network);
```
Comment (chillvybz) [inline src/client/js/utils/tokenList.js:50]: 👍

File: src/client/js/utils/tokenList.js [hunk 7]
```diff
@@ -59,6 +57,16 @@ window.TokenListManager = {
     });
   },
 
+  updateTokenListwithCustom: function (network) {
```
Comment (chillvybz) [inline src/client/js/utils/tokenList.js:60]: Keep this method as its' good.

But add a new method which I mentioned abvoe:

```
addCustomToken: function(token) {
    // saves into store().
   // then call updateTokenListWithCustom()
}
```

File: src/client/js/utils/wallet.js (hunk 1 above)
Comment (davidkoller318) [reply src/client/js/utils/wallet.js:113]: Okay

File: src/client/js/utils/tokenList.js (hunk 7 above)
Comment (davidkoller318) [reply src/client/js/utils/tokenList.js:60]: Okay.

File: src/client/css/custom/swap.scss (hunk 3 above)
Comment (davidkoller318) [reply src/client/css/custom/swap.scss]: Okay.

File: src/client/js/utils/wallet.js (hunk 2 above)
Comment (davidkoller318) [reply src/client/js/utils/wallet.js]: Okay

**Comment by davidkoller318** [comment]
Hi, @chillvybz 
fixed issue.

**Comment by chillvybz** [comment]
New updates are good. Please fix conflicts and do some light regression testing if the code still works. Thanks

**Comment by davidkoller318** [comment]
> New updates are good. Please fix conflicts and do some light regression testing if the code still works. Thanks

Okay

**Comment by davidkoller318** [comment]
Hi, @chillvybz . Resolved conflicts and it works properly.
Thank you.

**Review by chillvybz — APPROVED** [review]

**Pull request merged**